@here/harp-mapview/lib/MapView.ts

```diff
@@ -3294,4 +3332,49 @@ export class MapView extends THREE.EventDispatcher {
             return { width: clientWidth, height: clientHeight };
         }
     }
+
+    private limitPitch(pitchDeg: number, zoomLevel = this.zoomLevel) {
+        // 90 leads to imprecision issues.
+        let limitedPitch = Math.max(Math.min(MAX_PITCH, pitchDeg), 0);
+        if (this.projection.type === ProjectionType.Spherical) {
+            const maxPitchRadWithCurvature = Math.asin(
+                EarthConstants.EQUATORIAL_RADIUS /
+                    (MapViewUtils.calculateDistanceToGroundFromZoomLevel(this, zoomLevel) +
+                        EarthConstants.EQUATORIAL_RADIUS)
+            );
+            const maxPitchDegWithCurvature = THREE.Math.radToDeg(maxPitchRadWithCurvature);
+            limitedPitch = Math.min(limitedPitch, maxPitchDegWithCurvature);
+        }
+        return limitedPitch;
+    }
+
+    /*
+     * Return the map View camera target point in world space. If the camera currently not looking
+     * to the map surface returns a vector in the camera looking direction the away from the camera
+     * by the last lookAtDistance value.
+     */
+    private getTarget() {
+        this.camera.updateMatrixWorld(true);
+        let target: THREE.Vector3 | undefined;
+
+        if (this.elevationProvider !== undefined) {
+            const rect = this.canvas.getBoundingClientRect();
+
+            target = this.elevationProvider.rayCast(
+                rect.left + rect.width / 2,
+                rect.top + rect.height / 2
+            );
+        }
+
+        if (target === undefined) {
+            target = MapViewUtils.rayCastWorldCoordinates(this, 0, 0) || undefined;
```

Comment:
Maybe we should never do the raycasting but always use the formula below. Otherwise it could happen that distance between target and camera is bigger than m_lookAtDistance (especially if tilt > 60°)